pkg/lokifrontend/frontend/v2/frontend_scheduler_worker.go

```diff
@@ -325,6 +327,7 @@ func (w *frontendSchedulerWorker) schedulerLoop(loop schedulerpb.SchedulerForFro
 			}
 
 		case reqID := <-w.cancelCh:
+			fmt.Println("Got the cancel as well", reqID)
```

Comment:
Still keeping the comments, in case reviewer wants to run the test with and without fix and easy to understand where cancel is getting lost.

Will do cleanup after initial review